### ISSUE TITLE
add `Credo.Check.Readability.NoCaptureOperator`

### ISSUE THREAD

**Issue by markmark206** [rrrene/credo#1215]
(Thank you for this super-neat tool!;)

Please add a check (off by default, low priority) that would generate a warning on capture operator.

Reasoning:

Some teams might prefer explicit anonymous functions for ease of "at a glance" readability.

While the capture operator allows for shorter code, it adds cognitive overhead (and possibility of mis-reading) for the person quickly reading the code.

Code Examples:

1. the "capture operator" version

```elixir
current_execution.values
|> Enum.map(& &1.node_name)
```
(this code would be flagged by `Credo.Check.Readability.NoCaptureOperator`)

2. the "anonymous function" version

```elixir
current_execution.values
|> Enum.map(fn value -> value.node_name end)
```

Option 2 is obviously more verbose and not as clever, but it is also more explicit, and easier to correctly understand at a glance by developers even relatively new to Elixir, which is important for some teams.

**Comment by rrrene** [comment]
I am not against this per se, but I have a feeling that we want to be able to allow your example:

```elixir
Enum.map(values, & &1.node_name)
```
or
```elixir
Enum.map(values, &MyModule.fun/1)
```

but disallow more complex cases like

```elixir
Enum.map(values, & [&1, Math.sqrt(&1)])
```
or
```elixir
Enum.map(values, &Mod.puts(:stderr, &1))
```

WDYT?

**Comment by jmalovera10** [comment]
I think we may want this rule to only allow the following forms:

```elixir
Enum.map(values, &FunctionModule.increase/1)

Enum.map(values, &FunctionModule.increase(&1, 2))

Enum.map(values, my_func)
```

My argument is that you force that the explicit behavior is declared and enclosed in its own function block, hence making it easier for you to make sense of what the code is doing. It also helps readability because you are not making inline business logic, which could happen if you allow anonymous functions or standalone capture operations. 

Would it make sense?

**Comment by rrrene** [comment]
Thanks again for suggesting this 😀  It should be working on `master`.

You can try this by setting the Credo dep to

```elixir
{:credo, github: "rrrene/credo"}
```

Please report back if your "issue" is solved! 👍
